suresh/array-str-methods.js

```diff
@@ -0,0 +1,97 @@
+var array = ['a', 1, 2, 'b', 'xyz'];
+
+function indexOf(array, ele) {
+  for (var i = 0; i < array.length; i++) {
+    if (array[i] === ele) {
+      return i;
+    } 
+  }
+  return "element is not present"
+}
+
+console.log('Array - indexOf: ' + indexOf(array, 'b'));
+
+var str = 'learning coding'
+
+console.log('String - indexOf: ' + indexOf(str, 'a'));
+
+var str = 'I%am%Learning%coding';
+
+function split(string, delimiter) {
+  var newArray = [];
+  var index = 0;
+  var spacing = [];
+  var wordIndex = 0;
+  
+  for (var i = 0; i < string.length; i++) {
+    if (string[i] === delimiter) {
+      spacing[index] = i;
+      index++
+    }
+  }
+  
+  console.log('spacing', spacing);
+  var startIndex = 0;
+  
+  for (var j = 0; j <= spacing.length; j++) {
```

Comment:
This works but you are creating many conditions that it has to check through. Try to do this with less iterations. Store your value as you iterate the first time.